rust-lang/cargo
# refactor: migrate to `tracing`

### What does this PR try to resolve?

The migrates `CARGO_LOG` logging from `log` to `tracing`.

`tracing` is already widely used among the ecosystem and rustc adopted it for a while. We've chatted about this a while back in the weekly meeting.

Last week in the office hour, epage, Muscraft, and arlosi have no objection to this when I mentioned this upcoming PR. ehuss seems to looking forward to the migration <https://github.com/rust-lang/cargo/pull/12445#issuecomment-1664753017>. Hence posted this pull request.

### How should we test and review this PR?

By commits. And please check if `CARGO_LOG` works for you.

Largely, a search & replace worked just fine. Only in `util/network/http.rs` I did a manual fix.

The format is also a bit different. Should we make it compatible? (I don't want to though)

```apachelog
# before
[2023-08-07T10:53:48Z DEBUG cargo::sources::registry::http_remote] loading config
# after
2023-08-07T10:43:47.867106Z DEBUG cargo::sources::registry::http_remote: loading config
```

By default `log` enables all levels, whereas the built-in `fmt` subscriber in tracing-subscriber [enables up to `INFO` only](https://docs.rs/tracing-subscriber/0.3/tracing_subscriber/fmt/struct.Subscriber.html#associatedconstant.DEFAULT_MAX_LEVEL). Is that a thing we need to consider? I do feel like INFO is a better default.


### Additional information

```
-2 dependencies
+11 dependencies
```

Last year during 1.63.0 release cargo only had ~180 deps, but today it doubles the number to ~337 🥲.
